pkg/bgpv1/manager/manager.go

```diff
@@ -1045,13 +1083,28 @@ func getRouterID(config *v2alpha1api.CiliumBGPNodeInstance, ciliumNode *v2api.Ci
 	if err != nil {
 		return "", fmt.Errorf("failed to parse Node annotations for instance %v: %w", config.Name, err)
 	}
-
 	routerID, err := annoMap.ResolveRouterID(asn)
+
+	// If there are no annotations about router-id, router-id will be allocated based on the allocation mode
 	if err != nil {
-		if nodeIP := ciliumNode.GetIP(false); nodeIP == nil {
-			return "", fmt.Errorf("failed to get CiliumNode IP %v: %w", nodeIP, err)
-		} else {
+		// check if the router ID allocation mode is set to default
```

Comment:
I'd suggest simplifying this code, too many `if...else` blocks.  For eg, it can be written something like this :

```
routerID, err := annoMap.ResolveRouterID(asn)
if err == nil {
   return routerID, nil
} 
// debug level log for anno err

if option.Config.BGPRouterIDAllocationMode == defaults.BGPRouterIDAllocationMode {
  nodeIP := ciliumNode.GetIP(false); nodeIP != nil {
    routerID = nodeIP.String()
  } else {
    routerID, err = calcRouterIDFromMacAddress()
    if err != nil {
      return "", err
    } 
  }
  return routerID, nil   
}
return "", fmt.Errorf("no router-id found")
```

At some point, we need to remove support for annotation based router-id configuration. But for now we can keep it as is.